Storage-models.md

```diff
@@ -0,0 +1,373 @@
+
+**IMPORTANT: You MUST get your model's deletion policy approved through a design doc before opening a PR.**
+
+The deletion policies are defined in the `DELETION_POLICY` enum in [`core/storage/base_model/gae_models.py`](https://github.com/oppia/oppia/blob/develop/core/storage/base_model/gae_models.py):
+
+* `KEEP`: The model should be kept for auditing or logging purposes. For example: `DeletedUserModel` or `SentEmailModel`.
+* `DELETE`: The model only belongs to one user, and should be deleted. For example: `CompletedActivitiesModel` or `LearnerPlaylistModel`.
+* `DELETE_AT_END`: The model only belongs to one user and should be deleted, but since the data in that model is relevant for the wipeout process, the model should only be deleted at the end after all other models are deleted. For example: `UserSettingsModel` or `UserAuthDetailsModel`.
+* `LOCALLY_PSEUDONYMIZE`: The model should be pseudonymized accounting for the local context. For example, `GeneralFeedbackThreadModel` is pseudonymized together with `GeneralFeedbackMessageModel` and `GeneralSuggestionModel`.
+* `PSEUDONYMIZE_IF_PUBLIC_DELETE_IF_PRIVATE`: The model should be pseudonymized if it is accessible by the general public (like published explorations) and deleted if it is not public. For example: `ExplorationRightsModel` or `ExpSummaryModel`.
+* `NOT_APPLICABLE`: The model is not related to user data at all. For example: `ClassifierTrainingJobModel` or `CommunityContributionStatsModel`.
+
+Your model class must provide a static `get_deletion_policy()` method that returns one of these deletion policies. For example:
+
+```python
+@staticmethod
+def get_deletion_policy() -> base_models.DELETION_POLICY:
+    return base_models.DELETION_POLICY.DELETE
+```
+
+#### Takeout policy
+
+Takeout is our program that lets users download their data. Takeout needs to know two things to let users export their data:
+
+* How is each model associated to users? For example, there is one `UserSettingsModel` per user, but a single `ExplorationRightsModel` may be shared across many users.
+
+  This question is answered by the `get_model_association_to_user()` static method, which returns one of the values of the enum `MODEL_ASSOCIATION_TO_USER` in [`core/storage/base_model/gae_models.py`](https://github.com/oppia/oppia/blob/develop/core/storage/base_model/gae_models.py). These values are:
+
+  * `ONE_INSTANCE_PER_USER`: There is a single instance of the model per user. An example of this would be the `UserSettingsModel` because each user has exactly one settings model, and each settings model is for exactly one user.
```

Comment:
```suggestion
  * `ONE_INSTANCE_PER_USER`: There is a single instance of the model per user and a single user for each model. An example of this would be the `UserSettingsModel` because each user has exactly one settings model, and each settings model is for exactly one user.
```